### PR TITLE
Use git ls-tree to build file list

### PR DESCRIPTION
**Pull request by nemchik**
Fixes #

## Proposed Changes

1. Use `git ls-tree` to build file list

### Author/Contributor
- [ ] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: lib/functions/buildFileList.sh [hunk 1]
```diff
@@ -113,28 +113,8 @@ function BuildFileList() {
...
-      -not -name "*.jpeg" \
-      -not -name "*.pdf" \
-      -not -name "*.png" \
-      -not -name "*.webp" \
-      -not -name "*.woff" \
-      -not -name "*.woff2" \
-      -not -name "*.zip" \
-      -type f \
-      2>&1 | sort)
+    debug "Populating the file list with:[git ls-tree -r --name-only \"${GITHUB_SHA}\"]"
```
Comment (ferrarimarco) [inline lib/functions/buildFileList.sh:116]: Is `GITHUB_SHA` set when running outside the GH Actions environment (i.e. when `RUN_LOCAL` is `true`?

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

**Comment by ferrarimarco** [comment]
@nemchik there's a fix in my branch for the issue you see in the log.

File: lib/functions/buildFileList.sh (hunk 1 above)
Comment (nemchik) [reply lib/functions/buildFileList.sh:116]: Good point about local. `GITHUB_SHA` and `GITHUB_WORKSPACE` would not be set.

Comment (nemchik) [reply lib/functions/buildFileList.sh]: Should be handled now with 783e0cb41ce7eb8205ebbacdbb9324c74d1bf006

**Comment by nemchik** [comment]
> @nemchik there's a fix in my branch for the issue you see in the log.

Merged the changes from your branch.

**Comment by nemchik** [comment]
Posting here so it's more visible:

074444a522032f8f6ac246a1441d62a4a69faa39 removes the `RenameTestFolder` function as its purpose was to move files out of a hidden folder that the `find` command would not see when building the file list. `git ls-tree` is able to see hidden files, so this is not needed anymore.

Additional side effects are that some other files that were hidden are now being linted and causing the deploy-DEV to fail. The `.dockerignore` file is currently not passing the JSCPD linter. I do not know much about that linter.

**Comment by admiralAwkbar** [comment]
@nemchik  i have no idea why we keep getting this error...
```
2021-02-23 15:41:45 [ERROR]   Found errors in [jscpd] linter!
2021-02-23 15:41:45 [ERROR]   Error code: 1. Command output:
------
Error: ENOTDIR: not a directory, scandir '/tmp/lint/.dockerignore'
    at Object.readdirSync (node:fs:1044:3)
    at readdir (/node_modules/@nodelib/fs.scandir/out/providers/sync.js:39:31)
    at Object.read (/node_modules/@nodelib/fs.scandir/out/providers/sync.js:12:12)
    at SyncReader.scandirSync [as _scandir] (/node_modules/@nodelib/fs.scandir/out/index.js:17:17)
    at SyncReader._handleDirectory (/node_modules/@nodelib/fs.walk/out/readers/sync.js:28:34)
    at SyncReader._handleQueue (/node_modules/@nodelib/fs.walk/out/readers/sync.js:23:18)
    at SyncReader.read (/node_modules/@nodelib/fs.walk/out/readers/sync.js:15:14)
    at SyncProvider.read (/node_modules/@nodelib/fs.walk/out/providers/sync.js:11:29)
    at ReaderSync.walkSync [as _walkSync] (/node_modules/@nodelib/fs.walk/out/index.js:19:21)
    at ReaderSync.dynamic (/node_modules/fast-glob/out/readers/sync.js:13:21) {
  errno: -20,
  syscall: 'scandir',
  code: 'ENOTDIR',
  path: '/tmp/lint/.dockerignore'
}
```

**Comment by nemchik** [comment]
@admiralAwkbar i think it's because the `.dockerignore` is a symlink to `.gitignore` which i think might be messing up JSCPD.

**Comment by admiralAwkbar** [comment]
@nemchik crazy i told the cleanup to remove it but its still there as well.. it hates me...

**Comment by admiralAwkbar** [comment]
@nemchik solved it.. bad find...

**Comment by nemchik** [comment]
> @nemchik solved it.. bad find...

Nice!

**Pull request merged**